dali/pipeline/operators/reader/reader_op.h

```diff
@@ -61,7 +61,12 @@ class DataReader : public Operator<Backend> {
         curr_batch_producer_(0),
         consumer_cycle_(false),
         producer_cycle_(false),
-        samples_processed_(0) {}
+        device_id_(-1),
```

Comment:
what's the value of `spec.GetArgument<int>("device_id")` in the case of CPU backend? Maybe we can read it always for simplicity. Just a thought

Reply:
The value is valid, no matter if this is GPU or CPU op. But we assume that, at least for the device guard that device_id == -1 is CPU and noop for that. Maybe in the future when we make DALI able to run without GPU device_id == -1 could be used from python as well to tell pipeline that no GPU will be used.